src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfAcrions = [];
+  let currentAction = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        let addAction = { ...currentAction };
+
+        addAction = Object.assign(addAction, action.extraData);
+
+        arrayOfAcrions.push(addAction);
+        currentAction = addAction;
+        break;
+
+      case 'removeProperties':
+        const removeAction = { ...currentAction };
+
+        for (const property of action.keysToRemove) {
+          delete removeAction[property];
+        }
+        arrayOfAcrions.push(removeAction);
```

Comment:
**Checklist pt.4**
You should move this step out of the switch statement to keep your code DRY. (**Please check suggestion at `line 37`**)

The same is with other cases